src/SecTester.Scan/Scan.cs

```diff
@@ -1,58 +1,115 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 using SecTester.Scan.Models;
 
 namespace SecTester.Scan;
 
-public class Scan : IDisposable
+public class Scan : IAsyncDisposable
 {
+  private static readonly TimeSpan DefaultPollingInterval = TimeSpan.FromSeconds(5);
+
   private static readonly IReadOnlyCollection<ScanStatus> ActiveStatuses =
-    new[]
-    {
-      ScanStatus.Pending, ScanStatus.Running, ScanStatus.Queued
-    };
+    new[] { ScanStatus.Pending, ScanStatus.Running, ScanStatus.Queued };
 
   private static readonly IReadOnlyCollection<ScanStatus> DoneStatuses =
-    new[]
-    {
-      ScanStatus.Disrupted, ScanStatus.Done, ScanStatus.Failed, ScanStatus.Stopped
-    };
+    new[] { ScanStatus.Disrupted, ScanStatus.Done, ScanStatus.Failed, ScanStatus.Stopped };
 
   private readonly ScanOptions _options;
-
+  private readonly ILogger _logger;
   private readonly Scans _scans;
-  private readonly ScanState _scanState = new(ScanStatus.Pending);
+  private volatile ScanState _scanState = new(ScanStatus.Pending);
 
-  public Scan(Scans scans, ScanOptions options)
+  public string Id { get; init; }
```

Comment:
Make property get-only

```suggestion
  public string Id { get; }
```